src/beamlime/logging/resources.py

```
@@ -0,0 +1,118 @@
+    """
+    Create the directory if it does not exist.
+    Raises an error if the desired path is an existing file.
+    """
+    import os
+
+    if os.path.exists(dir_path) and not os.path.isdir(dir_path):
+        raise FileExistsError(
+            f"{dir_path} is a file. " "It should either not exist or be a directory."
+        )
+    elif not os.path.exists(dir_path):
+        os.mkdir(dir_path)
+
+    return DirectoryCreated(True)
+
+
+LogFileName = NewType("LogFileName", str)
+
+
+@log_providers.provider
+def create_log_file_name(
+    time_stamp: TimeStamp,
+    prefix: LogFilePrefix = DefaultPrefix,
+    suffix: LogFileExtension = DefaultFileExtension,
+) -> LogFileName:
+    """Combine prefix, timestamp and suffix into a new file name."""
+    return LogFileName(f"{prefix}--{time_stamp}.{suffix}")
+
+
+FileHandlerBasePath = NewType("FileHandlerBasePath", Path)
+DefaultDirectoryStatus = DirectoryCreated(True)
+
+
+@log_providers.provider
+def create_log_file_path(
+    directory_ready: DirectoryCreated = DefaultDirectoryStatus,
+    *,
+    parent_dir: LogDirectoryPath,
+    file_name: LogFileName,
+) -> FileHandlerBasePath:
+    """
+    Create a log file path joining ``parent_dir`` and ``file_name``,
+    check if the file name already exists and returns the file name.
+    It will wait 0.001 second if the file name exists.
+
+    """
+    if directory_ready:
+        return FileHandlerBasePath(Path(parent_dir, file_name))
+    else:
+        raise ValueError("Directory should be ready first.")
+
+
+@contextmanager
+def hold_logging() -> Iterator[RLock]:
+    from . import _lock
+
+    _lock.acquire() if _lock else ...
+    try:
+        yield _lock
+    finally:
+        _lock.release() if _lock else ...
+
+
+def cleanup_file_handlers(logger: logging.Logger):
+    """Find file handlers that are connectected to
+    non-existing files and remove them from the logger."""
+    from os.path import exists
```

It probably should raise an error instead of just removing the handlers...